api/queries_repo.go

```diff
@@ -316,6 +326,60 @@ func GitHubRepo(client *Client, repo ghrepo.Interface) (*Repository, error) {
 	return InitRepoHostname(result.Repository, repo.RepoHost()), nil
 }
 
+func GitHubRepoWithMergeQueue(client *Client, repo ghrepo.Interface, baseBranch string) (*Repository, error) {
```

Comment:
this is a duplicate of `GitHubRepo` the only difference being the addition of the `baseBranch` argument which is used in the `mergeQueue` field in the request. I don't know if it makes sense to have a new function just to support a new argument, but also wasn't sure if it made sense to modify the callers to support this specific use case